src/modm/driver/radio/sx1276_impl.hpp

```diff
@@ -152,34 +152,43 @@ Sx1276<SpiMaster, Cs>::setSyncWord(uint8_t syncWord)
 // -----------------------------------------------------------------------------
 
 template <typename SpiMaster, typename Cs>
-modm::ResumableResult<void>
+modm::ResumableResult<bool>
 Sx1276<SpiMaster, Cs>::transmit(uint8_t* data, uint8_t length)
 {
 	RF_BEGIN();
 
-	RF_CALL(changeMode(ModemMode::STDBY));
+	lastTransmitResult = false;
 
-	RF_CALL(writeRegister(Sx1276Register::PayloadLength,length));
-	RF_CALL(writeRegister(Sx1276Register::FifoTxBaseAddr,0x00));
-	RF_CALL(writeRegister(Sx1276Register::FifoAddrPtr,0x00));
+	//read the mode register and check if we're in the RX Mode
+	opModeShadow.value = RF_CALL(readRegister(Sx1276Register::OpMode));
+	if(ModemMode_t::get(opModeShadow) == ModemMode::STDBY)
+	{
+		RF_CALL(writeRegister(Sx1276Register::PayloadLength,length));
+		RF_CALL(writeRegister(Sx1276Register::FifoTxBaseAddr,0x00));
+		RF_CALL(writeRegister(Sx1276Register::FifoAddrPtr,0x00));
 
-	RF_WAIT_UNTIL(this->acquireMaster());
+		RF_WAIT_UNTIL(this->acquireMaster());
 
-	buffer[0] = 0x80 | static_cast<uint8_t>(Sx1276Register::Fifo);
+		buffer[0] = 0x80 | static_cast<uint8_t>(Sx1276Register::Fifo);
 
-	Cs::reset();
+		Cs::reset();
 
-	RF_CALL(SpiMaster::transfer(buffer,nullptr,1));
-	RF_CALL(SpiMaster::transfer(data,nullptr,length));
+		RF_CALL(SpiMaster::transfer(buffer,nullptr,1));
+		RF_CALL(SpiMaster::transfer(data,nullptr,length));
 
-	if(this->releaseMaster())
+		if(this->releaseMaster())
+		{
+			Cs::set();
+		}
+
+		RF_CALL(changeMode(ModemMode::TX));
+	}
+	else
 	{
-		Cs::set();
+		MODM_LOG_ERROR<<"SX1276: Transmission failed as the modem is busy"<<modm::endl;
```

Comment:
I thought logging into a driver after MODM_LOG_ERROR was not desirable? @salkinium 

Ok, the driver depends on `modm:debug` :+1:

Reply:
It's not, but there isn't an alternative. It would be nice to completely remove/configure logging at compile-time if `modm:debug` isn't available, but we don't that yet.